File: src/medrelay/enums/resource_status.py
from enum import StrEnum


class ResourceStatus(StrEnum):
    AVAILABLE = "AVAILABLE"
    ASSIGNED = "ASSIGNED"
    EN_ROUTE = "EN_ROUTE"
    ON_SCENE = "ON_SCENE"
    TRANSPORTING = "TRANSPORTING"
    RETURNING = "RETURNING"
    OUT_OF_SERVICE = "OUT_OF_SERVICE"

    def label(self) -> str:
        return {
            ResourceStatus.AVAILABLE: "Available",
            ResourceStatus.ASSIGNED: "Assigned",
            ResourceStatus.EN_ROUTE: "En Route",
            ResourceStatus.ON_SCENE: "On Scene",
            ResourceStatus.TRANSPORTING: "Transporting",
            ResourceStatus.RETURNING: "Returning",
            ResourceStatus.OUT_OF_SERVICE: "Out of Service",
        }[self]

File: app/Enums/ResourceStatus.php
<?php

declare(strict_types=1);

namespace App\Enums;

enum ResourceStatus: string
{
    case Available = 'available';
    case Assigned = 'assigned';
    case EnRoute = 'en_route';
    case OnScene = 'on_scene';
    case Transporting = 'transporting';
    case Returning = 'returning';
    case OutOfService = 'out_of_service';

    public function label(): string
    {
        return match ($this) {
            self::Available => 'Available',
            self::Assigned => 'Assigned',
            self::EnRoute => 'En Route',
            self::OnScene => 'On Scene',
            self::Transporting => 'Transporting',
            self::Returning => 'Returning',
            self::OutOfService => 'Out of Service',
        };
    }
}
